src/bun.js/bindings/bindings.cpp

```diff
@@ -232,20 +269,24 @@ AsymmetricMatcherResult matchAsymmetricMatcher(JSGlobalObject* globalObject, JSC
 
         return AsymmetricMatcherResult::FAIL;
     } else if (auto* expectArrayContaining = jsDynamicCast<JSExpectArrayContaining*>(matcherPropCell)) {
+        if (!readFlagsAndProcessPromise(expectArrayContaining, flags, globalObject, otherProp))
+            return AsymmetricMatcherResult::FAIL;
+
         JSValue expectedArrayValue = expectArrayContaining->m_arrayValue.get();
 
-        if (JSC::isArray(globalObject, otherProp)) {
-            if (JSC::isArray(globalObject, expectedArrayValue)) {
-                JSArray* expectedArray = jsDynamicCast<JSArray*>(expectedArrayValue);
-                JSArray* otherArray = jsDynamicCast<JSArray*>(otherProp);
+        if (JSC::isArray(globalObject, expectedArrayValue)) {
+            JSArray* expectedArray = jsDynamicCast<JSArray*>(expectedArrayValue);
+            unsigned expectedLength = expectedArray->length();
 
-                unsigned expectedLength = expectedArray->length();
-                unsigned otherLength = otherArray->length();
+            if (expectedLength == 0) {
+                // jest makes it so an empty array matches anything,
+                // even a primitive like a string or a number
```

Comment:
What's an example of this? We have deviated a handfull of times from Jest where something was _obviously_ wrong. (aka. potential to break is outweighed by catching likely errored code)

Reply:
I added some tests to `expect.test.js`, and ran them on Jest where they pass
```js
    test("ArrayContaining matches", () => {
      expect(expect.arrayContaining([])).toEqual(42);
      expect(expect.arrayContaining([])).toEqual("jest");
      expect(expect.arrayContaining(["foo"])).toEqual(["foo"]);
      expect(expect.arrayContaining(["foo"])).toEqual(["foo", "bar"]);
      expect(expect.arrayContaining([])).toEqual({});
    });
```
Although I agree this is a very weird behavior

Reply:
Let's change those to fail, so `expect.arrayContaining` fails when not an array. Same with the other asymmetric matchers like `expect.objectContaining`. This just defies common sense, if code breaks from these paths, it was already broken.